scene/gui/split_container.cpp

```diff
@@ -213,52 +206,54 @@ void SplitContainer::_resort() {
+	int sep = _get_separation();
 
 	// Move the children, including the dragger.
 	if (vertical) {
-		fit_child_in_rect(first, Rect2(Point2(0, 0), Size2(get_size().width, middle_sep)));
-		int sofs = middle_sep + sep;
+		fit_child_in_rect(first, Rect2(Point2(0, 0), Size2(get_size().width, computed_split_offset)));
+		int sofs = computed_split_offset + sep;
 		fit_child_in_rect(second, Rect2(Point2(0, sofs), Size2(get_size().width, get_size().height - sofs)));
 	} else {
 		if (is_layout_rtl()) {
-			middle_sep = get_size().width - middle_sep - sep;
-			fit_child_in_rect(second, Rect2(Point2(0, 0), Size2(middle_sep, get_size().height)));
-			int sofs = middle_sep + sep;
+			computed_split_offset = get_size().width - computed_split_offset - sep;
+			fit_child_in_rect(second, Rect2(Point2(0, 0), Size2(computed_split_offset, get_size().height)));
+			int sofs = computed_split_offset + sep;
 			fit_child_in_rect(first, Rect2(Point2(sofs, 0), Size2(get_size().width - sofs, get_size().height)));
 		} else {
-			fit_child_in_rect(first, Rect2(Point2(0, 0), Size2(middle_sep, get_size().height)));
-			int sofs = middle_sep + sep;
+			fit_child_in_rect(first, Rect2(Point2(0, 0), Size2(computed_split_offset, get_size().height)));
+			int sofs = computed_split_offset + sep;
 			fit_child_in_rect(second, Rect2(Point2(sofs, 0), Size2(get_size().width - sofs, get_size().height)));
 		}
 	}
 
-	// Handle the dragger visibility and position.
-	if (dragger_visibility == DRAGGER_VISIBLE && !collapsed) {
-		dragging_area_control->show();
-
-		int dragger_ctrl_size = MAX(sep, theme_cache.minimum_grab_thickness);
-		if (vertical) {
-			dragging_area_control->set_rect(Rect2(Point2(0, middle_sep - (dragger_ctrl_size - sep) / 2), Size2(get_size().width, dragger_ctrl_size)));
-		} else {
-			dragging_area_control->set_rect(Rect2(Point2(middle_sep - (dragger_ctrl_size - sep) / 2, 0), Size2(dragger_ctrl_size, get_size().height)));
-		}
+	if (vertical) {
+		split_bar_rect = Rect2(drag_area_margin_begin, computed_split_offset, get_size().width - drag_area_margin_begin - drag_area_margin_end, sep);
+	} else {
+		split_bar_rect = Rect2(computed_split_offset, drag_area_margin_begin, sep, get_size().height - drag_area_margin_begin - drag_area_margin_end);
+	}
 
-		dragging_area_control->queue_redraw();
+	dragging_enabled ? dragging_area_control->show() : dragging_area_control->hide();
+	const int dragger_ctrl_size = MAX(sep, theme_cache.minimum_grab_thickness);
+	if (vertical) {
+		dragging_area_control->set_rect(Rect2(drag_area_margin_begin,
+				computed_split_offset - (dragger_ctrl_size - sep) * 0.5 + drag_area_scrollbar_offset,
+				get_size().width - drag_area_margin_begin - drag_area_margin_end,
+				dragger_ctrl_size));
 	} else {
-		dragging_area_control->hide();
+		dragging_area_control->set_rect(Rect2(computed_split_offset - (dragger_ctrl_size - sep) * 0.5 + drag_area_scrollbar_offset,
```

Comment:
```suggestion
		dragging_area_control->set_rect(Rect2(computed_split_offset - (dragger_ctrl_size - sep) * 0.5 + drag_area_scrollbar_offset * (is_layout_rtl() ? -1 : 1),
```

This should also be reversed in rtl, `drag_area_scrollbar_offset` should increase when moving toward the second split.

Reply:
Is this standard for rtl, that every kind of horizontal offset should reverse? I'll assume it is and implement this also.

Reply:
When switching to/from rtl it should look mirrored, but we need to be careful that stuff like input still works as expected.